### PR TITLE
Log github API call failures

### PR DESCRIPTION
**Pull request by Limess**
Currently Github API failures are logged as destructuring errors, as the response is not in the expected array/object format. Anything is treated as a success.

This change propagates the child loggers to the github API caller, which logs an error and throws on any non ok status code.

Example error - which I believe implies that the returned value is not an array so is not iterable:

```
"error":{"type":"TypeError","message":"(intermediate value) is not iterable","stack":"TypeError: (intermediate value) is not iterable\n    at path (/var/task/lambdas/ingester/src/webpack:/lambdas/ingester/src/releaseLogHandler.js:110:9)\n    at process._tickCallback (internal/process/next_tick.js:68:7)"}
```

### CODE REVIEW

File: lambdas/ingester/src/releaseLogHandler.js [hunk 1]
```diff
@@ -66,69 +66,97 @@ const parseRecord = childLogger => record => {
 	};
 };
 
-const githubAPI = (path, { previewMode, ...options } = {}) => {
+const createGithubAPI = (log = logger) => async (
```
Comment (wheresrhys) [inline lambdas/ingester/src/releaseLogHandler.js]: createGithubAPI is a bit of a weird name. Maybe createGitApiClient?

File: lambdas/ingester/src/releaseLogHandler.js [hunk 2]
```diff
@@ -66,69 +66,97 @@ const parseRecord = childLogger => record => {
...
-};
+	if (!response.ok) {
+		const contentType = response.headers.get('content-type');
+		const body = await (contentType &&
+		contentType.includes('application/json')
+			? response.json()
+			: response.text());
+
+		log.error({
+			event: 'GITHUB_API_FAILURE',
```
Comment (wheresrhys) [inline lambdas/ingester/src/releaseLogHandler.js]: also log the url and other params sent to the api?

Comment (Limess) [reply lambdas/ingester/src/releaseLogHandler.js]: Good idea, added logging of the URL and all options aside from the Authorization header.

File: lambdas/ingester/src/releaseLogHandler.js (hunk 1 above)
Comment (Limess) [reply lambdas/ingester/src/releaseLogHandler.js]: Added `Client` as a suffix.

**Pull request merged**